dev/ScrollViewer/ScrollViewerBringIntoViewOperation.h

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#pragma once
+
+#include "common.h"
+
+class ScrollViewerBringIntoViewOperation
+{
+public:
+    ScrollViewerBringIntoViewOperation(winrt::UIElement const& targetElement);
+    ~ScrollViewerBringIntoViewOperation();
+
+    bool HasMaxTicksCount() const;
+    winrt::UIElement TargetElement() const;
+    int8_t TicksCount() const;
+    int8_t TickOperation();
+
+private:
+    // Number of UI thread ticks allowed before this expected bring-into-view operation is no
+    // longer expected and removed from the ScrollViewer's m_bringIntoViewOperations list.
+    static constexpr int8_t s_maxTicksCount{ 3 };
```

Comment:
>static constexpr int8_t s_maxTicksCount{ 3 }; [](start = 4, length = 45)

This seems fragile, and it seems like we are building up a few cases where we depend on timings of ticks to get the right behavior. I don't have a better solution in mind but each time I see one of these patches I get more nervous that we aren't taking the correct approach.

Reply:
Totally agree. The right fix long term is to extent the FocusManager class so that AnimationDesired becomes a parameter for the focus change requests. I still need to make sure a feature request is opened for that based on an email thread I had with Austin/Micah/Miguel. For now, this was the best workaround I could come up with.